mypyc/test-data/run-loops.test

```diff
@@ -452,3 +452,19 @@ def bar(x: Optional[str]) -> None:
 [file driver.py]
 from native import bar
 bar(None)
+
+[case testRangeObject]
+r1 = range(4, 12, 2)
```

Comment:
Can you change this to use a `test_` function without a custom driver, as that's the "modern" way to write tests.

Test also that coercion to `range` works, for example something like this:

```
def f(r: range) -> None:
    pass

r: Any = range(...)
f(r)  # OK

r = 'x'
f(r)  # Should be a TypeError

ff: Any = f
ff(r)  # Again should be a TypeError; this uses the wrapper function since the callee has type Any
```

Reply:
I got this error after adding function `def f(r: range) -> None:`
```
run-loops.test:460: error: Function "builtins.range" is not valid as a type
run-loops.test:460: note: Perhaps you need "Callable[...]" or a callback protocol?
```

Reply:
You'll need to change the definition of `range` into a class in `mypyc/test-data/fixtures/ir.py`.